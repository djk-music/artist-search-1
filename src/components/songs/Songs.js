import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { getArtistWorks } from '../../services/artistsAndWorks';
import Song from './Song';

export default class Songs extends PureComponent {
  state = {
    songs: []
  }

  static propTypes = {
    match: PropTypes.object.isRequired
  }

  componentDidMount() {
    getArtistWorks(this.props.match.params.id)
      .then(response => {
        this.setState({ songs: response.results });
      });
  }

  render() {
    const { songs } = this.state;
    const listOfSongs = songs.map(song => {
      const songTitle = song.title;
      songTitle.replace(/\s+/g, '-').toLowerCase();
      return <Link to={`/songs/${this.props.match.params.name}/${songTitle}`} key={song.id}><li><Song song={song}/></li></Link>;
    });
    
    return (
      <Fragment>
        <h2>Songs</h2>
        <div>
          <h3>songs</h3>
          <ul>
            {listOfSongs}
          </ul>
        </div>
      </Fragment>
    );
  }
}
