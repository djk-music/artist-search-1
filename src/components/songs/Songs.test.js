import React from 'react';
import renderer from 'react-test-renderer';
import Songs  from './Songs';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../services/artistsAndWorks.js');

describe('Songs', () => {
  it('matches a snapshot', () => {
    const match = {
      params: {
        id: '2344'
      }
    };
    const tree = renderer.create(
      <MemoryRouter><Songs match={match}/></MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
