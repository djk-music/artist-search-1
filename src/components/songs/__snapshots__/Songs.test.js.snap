// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs matches a snapshot 1`] = `
Array [
  <h2>
    Songs
  </h2>,
  <div>
    <h3>
      songs
    </h3>
    <ul />
  </div>,
]
`;
